pandas/core/groupby.py

```diff
@@ -1950,7 +1950,12 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
 
                 # fix bug #GH8868 sort=False being ignored in categorical groupby
                 else:
-                    self.grouper = self.grouper.reorder_categories(self.grouper.unique())
+                    from pandas.core.nanops import unique1d
+                    # take unique codes preserving its original order
+                    unique_codes = unique1d(self.grouper.codes)
+                    # do not coerce to np.array to make datetime-like work
+                    cat = self.grouper.categories.take(unique_codes)
+                    self.grouper = self.grouper.reorder_categories(cat)
```

Comment:
I think unique should always return a Categorical

(like above) - wonder why we don't have that